server/doc.js

```diff
@@ -0,0 +1,36 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
+  title: {type: String, required:true},
+  content: {type: String, required:true},
+  owner_id: {type: Schema.Types.ObjectId, ref: 'User'},
+  locked: Boolean,
+  revisions:[
+    {type: Schema.Types.ObjectId, ref: 'Revision'}
+  ],
+  last_revised: {type: Date, 
+    default: Data.now
+  }
+})
+
+docSchema.methods.filter = function(taboo_words){
+  //A String is considered a word if there is a space
+  //Checks all the words and see if any are in taboo words and replaces those words
+  //returns indexes of all taboo words, format at each index: [start, end + 1]
+  idx_list = [];
+  for(let i = 0; i < taboo_words.length; i++){
```

Comment:
since you are not assigning values to the indexes of taboo_words, use an iterative for loop, i.e 
for(const taboo of taboo_words){
}

Reply:
Please address this

Reply:
It was changed previously already